src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -0,0 +1,146 @@
+
+            if (newCategory.getBudget() < 0) {
+                // 2. Category budget less than 0 fail
+                CategoryOD categoryODFailAdd = new CategoryOD("Category budget can't be less than $0. Please try again!");
+                return categoryOB.fail(categoryODFailAdd);
+            }
+            month.addCategory(newCategory);
+
+            CategoryOD categoryODSuccessAdd = new CategoryOD("You have added a new category!");
+            return categoryOB.success(categoryODSuccessAdd);
+
+        } catch(NumberFormatException|NullPointerException e){
+            //1. NumberFormatException/NullPointerException fail
+            CategoryOD categoryODFailAdd = new CategoryOD("Category budget is needs to be a number. Please try again!");
+            return categoryOB.fail(categoryODFailAdd);
+
+        } catch (EntityException e) {
+            //3. EntityException fail
+            CategoryOD categoryODFailAdd = new CategoryOD("There is already a category with this new name in this month.");
+            return categoryOB.fail(categoryODFailAdd);
+        }
+    }
+    /**
+     * Overrides method in CategoryIB.
+     * Attempts to edit a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition listed below:
+     *  1. Repeated Name: User tries to edit category name to another name that exists in the month.
+     *  2. NoSuchElementException: User tries to edit a category that does not exist.
+     *  3. NumberFormatException: User tries to edit a budget value with input that can not be converted to a double.
+     *  4. Category budget less than 0: User tries to edit a budget value with input that is a negative number.
+     *
+     * @param categoryIDEdit CategoryID required for editing a new category to the designated monthID MonthlyStorage Object.
+     * @return CategoryOD String message indicating success/fail add attempt.
+     */
+    @Override
+    public CategoryOD editCategoryInMonth(CategoryID categoryIDEdit) throws EntityException {
+        MonthlyStorage month = categoryIDEdit.getSession().getMonthlyData(categoryIDEdit.getMonthID());
+        ArrayList<Category> categoryList = month.getCategoryData();
+        try {
+            double valueDouble = toDouble(categoryIDEdit.getValue());
+            if (valueDouble < 0) {
+                //4. Category budget less than 0 fail
+                CategoryOD categoryODFailEdit = new CategoryOD("Category budget can't be less than $0. Please try again!");
+                return categoryOB.fail(categoryODFailEdit);
+            }
+
+            Category existingCategory = findCategory(categoryList, categoryIDEdit.getOldCategory());
+
+            for (Category category1 : categoryList) {
+                if (Objects.equals(categoryIDEdit.getName(), categoryIDEdit.getOldCategory())){
```

Comment:
This if statement looks like it doesn't need to be run inside the loop. Also, could you add a short comment explaining what it's doing, cause it's a bit confusing

Reply:
That statement is not needed. Good eye.

Reply:
It seems like the problem was that two input data values were compared. I saw the updated code was comparing correctly between categoryIDEdit.getName() and name of all categories stored, so I will resolved this conversation.